api/envoy/service/endpoint/v3/leds.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto3";
+
+package envoy.service.endpoint.v3;
+
+import "envoy/service/discovery/v3/discovery.proto";
+
+import "google/api/annotations.proto";
+
+import "envoy/annotations/resource.proto";
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.service.endpoint.v3";
+option java_outer_classname = "LedsProto";
+option java_multiple_files = true;
+option java_generic_services = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#not-implemented-hide:]
+// [#protodoc-title: LEDS]
+// Locality-Endpoint discovery :ref:`architecture overview
+// <arch_overview_service_discovery_types_leds>`
+
+service LocalityEndpointDiscoveryService {
```

Comment:
Out of curiosity, is there any case where we actually need a separate LEDS service, or will all known use-cases of LEDS actually use ADS?

Reply:
My experience has been that endpoint data is different enough from regular config to merit special data plane treatment.  The reasons include that endpoint data often changes faster that the rest of the config and loading endpoint data presents different risks than those seen while loading standard config.  The most common failure modes when loading endpoint config is not validation errors but potential for excessive memory/cpu usage due to update size and frequency.

Endpoint distribution may also be one of the first possible targets for federation or sharding of control plane components in order to handle scaling better.